lib/net/wifi.toit

```diff
@@ -128,9 +136,17 @@ class WifiInterface_ extends SystemInterface_ implements Interface:
   constructor client/WifiServiceClient connection/List:
     super client connection
 
+  get_ap_info -> AccessPoint:
+    ap_info := (client_ as WifiServiceClient).get_ap_info handle_
+    return AccessPoint
+        --ssid=ap_info[WIFI_SCAN_SSID_]
+        --bssid=ap_info[WIFI_SCAN_BSSID_]
+        --rssi=ap_info[WIFI_SCAN_RSSI_]
+        --authmode=ap_info[WIFI_SCAN_AUTHMODE_]
+        --channel=ap_info[WIFI_SCAN_CHANNEL_]
+
   signal_strength -> float:
-    rssi := (client_ as WifiServiceClient).rssi handle_
-    if not rssi: throw "wifi not connected in STA mode"
+    rssi := get_ap_info.rssi
```

Comment:
Maybe save the allocation of the `AccessPoint` instance?

```suggestion
    info := (client_ as WifiServiceClient).get_ap_info handle_
    rssi := info[WIFI_SCAN_RSSI_]
```

Reply:
Getting value directly looks better.